content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,88 @@
+
+Another important aspect of DAGs to note: if you change a small part of a DAG, the whole root will become different. Take a look at [this helpful illustration using bananas](https://media.consensys.net/ever-wonder-how-merkle-trees-work-c2f8b7100ed3) from our friends at Consensys.
+
+### 3 \\ The DHT
+
+<img alt="IPFS Stack - Routing and Exchange" src="../assets/ipfs_stack-exchange_routing.png" width="200px" />
+
+So, to recap, IPFS lets you to give CIDs to content, and link that content together by generating DAGs using IPLD. Now let's move on to the last piece: how you find and move that content around!
+
+In order to find which peer has certain content (_discovery_), you use a [_distributed hash table_](https://en.wikipedia.org/wiki/Distributed_hash_table), or DHT. **IPFS is essentially a database.** Just like key-value pairs, if you have a peer, it will let you find content, and if you have content, it will let you find you the peer where the content is stored.
+
+The <a hrefm src="https://libp2p.io/">libp2p project</a> is the part of the IPFS ecosystem that provides the DHT and handles peers connecting and talking to each other. It can be used as a tool for other distributed systems as well; just as there is "IPFS-flavored IPLD," there's also "IPFS-flavored libp2p."
+
+Once you know where your content is (ie, which peer or peers are storing the various blocks that make up the content you're after), you use the DHT **again** to find the current location of those peers (_routing_). So, in order to get to content, you use libp2p to query the DHT twice.
+
+You've discovered your content, and you've found the current location(s) of that content — now you need to connect to that content (_exchange_). libp2p again to the rescue! _[Maybe something here about bootstrapping, Bitswap, or Graphsync.]_
+
+What makes the IPFS protocol especially useful for peer to peer connections is _multiplexing_. Traditionally, you open a different connection to each protocol you talk to. Using IPFS, you open just one connection, and you multiplex everything on that. For everything your peers have to talk to each other about, you send a little bit of each thing, and the other end knows how to sort those chunks where they belong.
+
+This is useful because establishing a connection in p2p systems is sometimes difficult depending on where they are. Once you have that connectivity, you can do whatever you need on it. And you only need one socket.
```

Comment:
```suggestion
This is useful because connections are usually hard to set up and expensive to maintain. Once you have that connectivity, you can do whatever you need on it.
```

Reply:
Done, with minor copyedits.